### PR TITLE
feat: add xmarket es dataset

### PR DESCRIPTION
**Pull request by guenthermi**
This PR adds the Spanish version of the [XMarket dataset](https://xmrec.github.io/) which is uploaded on HF [here](https://huggingface.co/datasets/jinaai/xmarket_es)

### CODE REVIEW

**Comment by guenthermi** [comment]
| Model  |   ndcg_at_1 |   ndcg_at_3 |   ndcg_at_5 |   ndcg_at_10 |   ndcg_at_100 |   ndcg_at_1000 |   map_at_1 |   map_at_3 |   map_at_5 |   map_at_10 |   map_at_100 |   map_at_1000 |   recall_at_1 |   recall_at_3 |   recall_at_5 |   recall_at_10 |   recall_at_100 |   recall_at_1000 |   precision_at_1 |   precision_at_3 |   precision_at_5 |   precision_at_10 |   precision_at_100 |   precision_at_1000 |   mrr_at_1 |   mrr_at_3 |   mrr_at_5 |   mrr_at_10 |   mrr_at_100 |   mrr_at_1000 |   evaluation_time |
|:----------------------|------------:|------------:|------------:|-------------:|--------------:|---------------:|-----------:|-----------:|-----------:|------------:|-------------:|--------------:|--------------:|--------------:|--------------:|---------------:|----------------:|-----------------:|-----------------:|-----------------:|-----------------:|------------------:|-------------------:|--------------------:|-----------:|-----------:|-----------:|------------:|-------------:|--------------:|------------------:|
| e5 multilingual base  |     0.12084 |     0.12792 |     0.12946 |      0.13578 |       0.17323 |        0.21613 |    0.03683 |    0.05673 |    0.06423 |     0.07338 |      0.0901  |       0.09488 |       0.03683 |       0.0747  |       0.09569 |        0.13311 |         0.29535 |          0.49928 |          0.12084 |          0.10601 |          0.09236 |           0.07483 |            0.02761 |             0.0077  |    0.12084 |    0.17044 |    0.18158 |     0.19091 |      0.19972 |       0.20045 |             41.86 |
| e5 multilingual large |     0.13259 |     0.13206 |     0.13501 |      0.1413  |       0.1821  |        0.22974 |    0.03765 |    0.05874 |    0.0671  |     0.07655 |      0.09423 |       0.09943 |       0.03765 |       0.07753 |       0.10098 |        0.13988 |         0.31553 |          0.53681 |          0.13259 |          0.10807 |          0.09611 |           0.07692 |            0.02836 |             0.00829 |    0.13259 |    0.17874 |    0.19087 |     0.20073 |      0.20973 |       0.21049 |            125.37 |

**Review by violenil — APPROVED** [review]
LGTM!

**Review by jupyterjazz — APPROVED** [review]
LGTM!

File: mteb/tasks/Retrieval/XMarketESRetrieval.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        query_rows = datasets.load_dataset(
+            self.description["hf_hub_name"],
+            "queries",
+            languages=['es'],
+            split=self._EVAL_SPLIT,
+            trust_remote_code=True,
+        )
+        corpus_rows = datasets.load_dataset(
+            self.description["hf_hub_name"], "corpus", languages=['es'], split=self._EVAL_SPLIT, trust_remote_code=True
```
Comment (jupyterjazz) [inline mteb/tasks/Retrieval/XMarketESRetrieval.py]: maybe it would be nicer to replace languages=['es'] with languages=kwargs['eval_langs']

Comment (jupyterjazz) [reply mteb/tasks/Retrieval/XMarketESRetrieval.py]: or self.description['eval_langs']

not sure which is correct

Comment (guenthermi) [reply mteb/tasks/Retrieval/XMarketESRetrieval.py]: I think eval_langs is supposed to be the set of supported languages. There is probably another kwargs in in load_data for the languages. I can look into it

**Pull request merged**